src/types.ts

```diff
@@ -12,8 +14,21 @@ export interface SmartweaveContractSource {
 }
 
 export type SearchBarProps = {
-  buttonAction: () => void;
+  predicate: (value: string) => boolean;
   placeholderText: string;
-  headerText: string;
-  footerText: string;
+  headerElement: JSX.Element;
+  footerElement: JSX.Element;
+  values: string[] | object;
+};
+
+export type SearchBarHeaderProps = {
+  defaultText: string;
+  isValid?: boolean;
+  isDefault?:boolean;
+  text?: string;
+};
+
+export type SearchBarFooterProps = {
+  defaultText: string;
+  isValid?: boolean;
```

Comment:
let's add `isAvailable` here given the design implications for what should be displayed based on when an ANT is taken (`isAvailable` is false, vs `isValid`), we might want to add these both to the header too. we can kick this to another PR